### PR TITLE
fix concurrent player check

### PR DESCRIPTION
**Pull request by windo**
The `player_pool` property does not exist on main site (yet?).

### CODE REVIEW

**Comment by windo** [comment]
Actually, even with this, I still got:

```
May 12 16:44:07 ! source-map-support.js:495 
May 12 16:44:07 ! source-map-support.js:496 /home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:712
            const game_count = Object.keys(this.connected_games).filter((game_id) => {
                                                                 ^
May 12 16:44:07 ! source-map-support.js:499 TypeError: Cannot read properties of null (reading 'player_pool')
    at /home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:717:27
    at Array.filter (<anonymous>)
    at Main.checkGamesPerPlayer (/home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:712:66)
    at Main.handleNotification (/home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:294:30)
    at GobanSocket.<anonymous> (/home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:150:58)
    at GobanSocket.emit (/home/siim/projects/gtp2ogs-release/node_modules/eventemitter3/index.js:181:35)
    at WebSocket.<anonymous> (/home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/node_modules/goban/src/GobanSocket.ts:266:22)
    at callListener (/home/siim/projects/gtp2ogs-release/node_modules/ws/lib/event-target.js:290:14)
    at WebSocket.onMessage (/home/siim/projects/gtp2ogs-release/node_modules/ws/lib/event-target.js:209:9)
    at WebSocket.emit (node:events:517:28)
```

**Comment by raylu** [comment]
oof, https://github.com/online-go/goban/blob/7966731fea970341987103a9e02e38c2bd31255c/src/GoEngine.ts#L120 marks `player_pool` as optional but `tsc` doesn't error because https://github.com/online-go/gtp2ogs/blob/9478fd1c808dcb23fbc6c6d6c5d6b1f7a5a07e62/tsconfig.json#L26

File: src/main.ts [hunk 1]
```diff
@@ -710,7 +710,14 @@ class Main {
         config;
         if (config.max_games_per_player) {
             const game_count = Object.keys(this.connected_games).filter((game_id) => {
-                return !!this.connected_games[game_id].state?.player_pool[player_id];
+                const state = this.connected_games[game_id]?.state;
+                if (state === undefined) {
+                    return false;
+                }
+                if (state.player_pool !== undefined) {
+                    return !!state.player_pool[player_id];
```
Comment (raylu) [inline src/main.ts:718]:
```
May 12 16:44:07 ! source-map-support.js:499 TypeError: Cannot read properties of null (reading 'player_pool')
    at /home/siim/projects/gtp2ogs-release/dist/webpack:/gtp2ogs/src/main.ts:717:27
```
means that `state` was null, not undefined
```suggestion
                if (state?.player_pool !== undefined) {
                    return !!state.player_pool[player_id];
```

**Comment by windo** [comment]
> oof, https://github.com/online-go/goban/blob/7966731fea970341987103a9e02e38c2bd31255c/src/GoEngine.ts#L120 marks `player_pool` as optional but `tsc` doesn't error because
> 
> https://github.com/online-go/gtp2ogs/blob/9478fd1c808dcb23fbc6c6d6c5d6b1f7a5a07e62/tsconfig.json#L26

I'm really not much of a javascript/typescript dev. Is there a neater way to express the count there, or is that OK?

File: src/main.ts (hunk 1 above)
Comment (windo) [reply src/main.ts]: Oh yeah! I had added a `trace.log` for when `state` was `undefined` but I guess it looks like that might be the case in pretty normal circumstances. I'll try to run your suggested form for a moment.

**Comment by windo** [comment]
I've been running with this change for a few days now, no more crashes so far.

**Pull request merged**